### PR TITLE
Access to HTTP POST body

### PR DESCRIPTION
**Pull request by dereulenspiegel**
This small change allows access to the body of a HTTP POST in case it hasn't been parsed as Form values. For the server to read the body and put in a char array, the content length has to be set and bigger than 0. Otherwise the body is omitted.

### CODE REVIEW

File: Sming/SmingCore/Network/HttpRequest.cpp [hunk 1]
```diff
@@ -28,19 +29,25 @@ HttpRequest::~HttpRequest()
...
+		String defaultValue /* = "" */)
 {
 	if (requestGetParameters && requestGetParameters->contains(parameterName))
-			return (*requestGetParameters)[parameterName];
+		return (*requestGetParameters)[parameterName];
 
 	return defaultValue;
 }
-String HttpRequest::getPostParameter(String parameterName, String defaultValue /* = "" */)
+String HttpRequest::getPostParameter(String parameterName,
```
Comment (anakod) [inline Sming/SmingCore/Network/HttpRequest.cpp]: Don't like that short lines, it's harder to read. No reason in actual monitors to fit in 80 chars.
Very long lines should be splited, of course.

**Comment by anakod** [comment]
Can you discover details of this P-R? Very many formating changes, what isn't good for understanding and code checking.

**Comment by dereulenspiegel** [comment]
Sorry for the formatting. Since Sming had an Eclipse config I simply used the configured formatting of the Eclipse project. I was under the impression the Eclipse project already had the correct formatting settings.
Basically I am reading the whole body into a char array in case it isn't parsed as form values and the content length is more than 0. In the destructor of the HTTP request object I am freeing up the memory again.
Could someone please provide the correct formatter settings? Then I will reformat the code.

**Comment by anakod** [comment]
> Sorry for the formatting. Since Sming had an Eclipse config I simply used the configured formatting of the Eclipse project. I was under the impression the Eclipse project already had the correct formatting settings.

Sorry, I'm not sure about specialized config, because I have never use it. May be just copy\paste required changes to original code?

**Comment by hreintke** [comment]
@dereulenspiegel : 
We are restructuring and cleaning Sming repositories and PR's. 
I am not sure about contents, functionality and formatting adaption as asked for by @anakod of this PR.
I would like to close this PR and ask you to submit a "clean" PR against develop.
Could you provide that ?

**Comment by hreintke** [comment]
Solved by #578

**Pull request closed without merging**